pennylane_lightning/core/src/utils/NDPermuter.hpp

```diff
@@ -189,11 +191,11 @@ template <size_t BLOCKSIZE = 1024> class DefaultPermuter {
             }
         }
 
-        std::vector<size_t> old_super_dimensions(num_indices, 1);
-        std::vector<size_t> new_super_dimensions(num_indices, 1);
+        std::vector<std::size_t> old_super_dimensions(num_indices, 1);
+        std::vector<std::size_t> new_super_dimensions(num_indices, 1);
 
         const std::size_t old_dimensions_size = shape.size();
-        for (size_t i = old_dimensions_size; --i;) {
+        for (std::size_t i = old_dimensions_size; --i;) {
```

Comment:
Do we need `;` here after `--i`?

Reply:
Yes, we need this as we don't have any explicit stopping condition here.